socorro/external/es/supersearch.py

```diff
@@ -22,6 +22,31 @@
 )
 
 
+def prune_invalid_indices(indices, policy, template):
+    """Prunes given indices by ones that were prior to the cutoff
+
+    :arg list indices: list of indices to prune
+    :arg int policy: number of days to retain indices--prior to this is the cutoff
+    :arg str template: index template
+
+    :returns: list of valid indices
+
+    """
+    now = utc_now()
+    marker = now - policy
+
+    # Generate the set of valid indexes between the cutoff and now.
+    valid_indices = set()
+    while marker <= now:
+        valid_indices.add(marker.strftime(template))
+        # NOTE(willkg): have to increment by 1 so we pick up week numbers for weeks
+        # that span a new year
+        marker = marker + datetime.timedelta(days=1)
```

Comment:
We could cache the set of valid indices for the day, but I didn't want to spend time thinking about testing the caching code. We can do that in a future PR.